### PR TITLE
Add TRTServer API to retrieve all unique models in model repositories

### PR DESCRIPTION
**Pull request by GuanLuo**
HTTP response body will look like:
```Protobuf
models {
  name: "graphdef_float32_float32_float32"
}
models {
  name: "savedmodel_float32_float32_float32"
}
models {
  name: "simple_graphdef_float32_float32_float32"
}
models {
  name: "simple_savedmodel_float32_float32_float32"
}
```

### CODE REVIEW

File: src/servers/grpc_server.cc [hunk 1]
```diff
@@ -648,6 +648,112 @@ StatusHandler::Process(Handler::State* state, bool rpc_ok)
...
+
+    TRTSERVER_ErrorDelete(err);
+
+    state->step_ = Steps::COMPLETE;
+    state->context_->responder_->Finish(response, grpc::Status::OK, state);
+  } else if (state->step_ == Steps::COMPLETE) {
+    state->step_ = Steps::FINISH;
+  }
+
+  // Only handle one status request at a time (to avoid having status
```
Comment (deadeyegoodwin) [inline src/servers/grpc_server.cc]: status -> model repository

File: src/servers/main.cc [hunk 2]
```diff
@@ -82,7 +82,8 @@ int32_t http_port_ = 8000;
 int32_t http_health_port_ = -1;
 std::vector<int32_t> http_ports_;
 std::vector<std::string> endpoint_names = {
-    "status", "health", "infer", "modelcontrol", "sharedmemorycontrol"};
+    "status",    "health", "infer", "modelcontrol", "sharedmemorycontrol",
```
Comment (deadeyegoodwin) [inline src/servers/main.cc]: Did the formatter add this extra whitespace?

File: src/core/trtserver.h [hunk 3]
```diff
@@ -768,6 +768,15 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ServerModelStatus(
     TRTSERVER_Server* server, const char* model_name,
     TRTSERVER_Protobuf** status);
 
+/// Get the index for all unique models in the model repository  as a
+/// TRTSERVER_Protobuf object. The caller takes ownership of the object
+/// and must call TRTSERVER_ProtobufDelete to release the object.
+/// \param server The inference server object.
+/// \param repository_index Returns the model repository index protobuf.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ModelRepositoryIndex(
```
Comment (deadeyegoodwin) [inline src/core/trtserver.h]: TRTSERVER_ServerModelRepositoryIndex

File: src/servers/main.cc (hunk 2 above)
Comment (GuanLuo) [reply src/servers/main.cc]: Yes

**Pull request merged**